src/dmd/dmodule.d

```diff
@@ -137,6 +137,74 @@ void semantic3OnDependencies(Module m)
         semantic3OnDependencies(m.aimports[i]);
 }
 
+/**
+ * Converts a chain of identifiers to the filename of the module
+ *
+ * Params:
+ *  packages = the names of the "parent" packages
+ *  ident = the name of the child package or module
+ *
+ * Returns:
+ *  the filename of the child package or module
+ */
+private const(char)[] getFilename(Identifiers* packages, Identifier ident)
+{
+    const(char)[] filename = ident.toString();
+
+    if (packages == null || packages.dim == 0)
+        return filename;
+
+    OutBuffer buf;
+    OutBuffer dotmods;
+    auto modAliases = &global.params.modFileAliasStrings;
+
+    void checkModFileAlias(const(char)[] p)
+    {
+        /* Check and replace the contents of buf[] with
+        * an alias string from global.params.modFileAliasStrings[]
+        */
+        dotmods.writestring(p);
+    Lmain:
+        foreach_reverse (const m; *modAliases)
+        {
+            printf("Mod: %s", m);
+            const q = strchr(m, '=');
+            assert(q);
+            if (dotmods.offset == q - m && memcmp(dotmods.peekString(), m, q - m) == 0)
+            {
+                buf.reset();
+                auto rhs = q[1 .. strlen(q + 1)];
+                printf("RHS: %s", rhs.ptr);
+                if (rhs.length > 0 && (rhs[$ - 1] == '/' || rhs[$ - 1] == '\\'))
+                    rhs = rhs[0 .. $ - 1]; // remove trailing separator
+                buf.writestring(rhs);
+                break Lmain; // last matching entry in ms[] wins
+            }
+        }
+        dotmods.writeByte('.');
+    }
+
+    foreach (pid; *packages)
+    {
+        const p = pid.toString();
+        buf.writestring(p);
+        if (modAliases.dim)
+            checkModFileAlias(p);
+        version (Windows)
+            enum FileSeparator = '\\';
+        else
+            enum FileSeparator = '/';
+        buf.writeByte(FileSeparator);
+    }
+    buf.writestring(filename);
+    if (modAliases.dim)
+        checkModFileAlias(filename);
+    buf.writeByte(0);
+    filename = buf.extractSlice();
+
+    return filename;
```

Comment:
This includes the trailing zero, should be `return filename[0..$-1];`

Reply:
Yeah; that would explain it. I probably should have read the source of `extractSlice`.